### PR TITLE
[feat] implement block_hash function

### PR DESCRIPTION
**Pull request by Gerson2102**
- [#40] issue #
- [X] follows contribution [guide](https://github.com/keep-starknet-strange/raito/blob/main/CONTRIBUTING.md)
- [X] code change includes tests

Adding the block hash function implementation. Function tested.

### CODE REVIEW

**Comment by Jeanmichel7** [comment]
last check should be after #103

File: src/types/block.cairo [hunk 1]
```diff
@@ -41,7 +42,54 @@ pub struct Header {
...
+        };
+        let merkle_root: Hash =
+            0xff104ccb05421ab93e63f8c3ce5c2c2e9dbb37de2764b3a3175c8166562cac7d_u256
+            .into();
+
+        let block_hash_result: Hash = block.header.hash(chain_state.best_block_hash, merkle_root);
+
+        //0x00000000d1145790a8694403d4063f323d499e655c83426834d4ce2f8dd4a2ee
+        let expected_block_hash: Hash =
+            0xeea2d48d2fced4346842835c659e493d323f06d4034469a8905714d100000000_u256
```
Comment (m-kus) [inline src/types/block.cairo:90]: I think it should be reverted here `0x00000000d1145790a8694403d4063f323d499e655c83426834d4ce2f8dd4a2ee_u256`

Reversed bytes (RPC): `00000000d1145790a8694403d4063f323d499e655c83426834d4ce2f8dd4a2ee`
Natural bytes: `eea2d48d2fced4346842835c659e493d323f06d4034469a8905714d100000000`
U256 little endian: `0x00000000d1145790a8694403d4063f323d499e655c83426834d4ce2f8dd4a2ee`

File: src/types/block.cairo [hunk 2]
```diff
@@ -41,7 +42,54 @@ pub struct Header {
...
+    use super::{Block, Header, BlockHash};
+    use raito::types::chain_state::ChainState;
+    use raito::utils::hash::Hash;
+
+    #[test]
+    fn test_block_hash() {
+        let mut chain_state: ChainState = Default::default();
+        chain_state
+            .best_block_hash =
+                0x55bd840a78798ad0da853f68974f3d183e2bd1db6a842c1feecf222a00000000_u256
```
Comment (m-kus) [inline src/types/block.cairo:73]: Needs to be reverted

File: src/types/block.cairo [hunk 3]
```diff
@@ -41,7 +42,54 @@ pub struct Header {
...
+            .into();
+        // block 170
+        let block = Block {
+            header: Header {
+                version: 1_u32, time: 1231731025_u32, bits: 0x1d00ffff_u32, nonce: 1889418792_u32
+            },
+            txs: ArrayTrait::new().span(),
+        };
+        let merkle_root: Hash =
+            0xff104ccb05421ab93e63f8c3ce5c2c2e9dbb37de2764b3a3175c8166562cac7d_u256
```
Comment (m-kus) [inline src/types/block.cairo:83]: Needs to be reversed

**Comment by m-kus** [comment]
@Gerson2102 please rebase and fix the tests, it should be ok now as #103 is merged

**Review by m-kus — APPROVED** [review]

**Pull request merged**